### PR TITLE
Make Buffer.read_only work properly with aullar

### LINKED ISSUE #155: Insertion into read-only buffer works
Basically, I noticed that, if you run an external command with Howl, you can insert text into the `ProcessBuffer`, even though it's read-only. Interestingly, you CANNOT replace text...because it's read-only!

This seems to be because [`howl.Editor.insert`](https://github.com/howl-editor/howl/blob/master/lib/howl/ui/editor.moon#L379) calls [`aullar.View.insert`](https://github.com/howl-editor/howl/blob/master/lib/aullar/view.moon#L310), which always calls [`aullar.Buffer.insert`](https://github.com/howl-editor/howl/blob/master/lib/aullar/buffer.moon#L157) instead of [`howl.Buffer.insert`](https://github.com/howl-editor/howl/blob/master/lib/howl/buffer.moon#L152). This skips the [check to make sure that the buffer is writable](https://github.com/howl-editor/howl/blob/master/lib/howl/buffer.moon#L296).

### PR DESCRIPTION
**Pull request by nilnor**
Moves read-only status down in to aullar.Buffer.

Fixes #155.

### CODE REVIEW

**Comment by shalabhc** [comment]
Cool - it works. But I get different messages for different operations:

- insert, paste or backspace: warning 'Attempt to modify read-only buffer'
- delete: error 'Attempt to modify read-only buffer *buffer-title*'
- select and cut: error '/lib/howl/ui/selection.moon:76: Attempt to modify read-only buffer'

Would be nice if all operations had the same warning as insert.

**Comment by nilnor** [comment]
@shalabhc Have a look now, should be more uniform. I also fixed a related problem with the ProcessBuffer.

File: bundles/vi/state.moon [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import bindings from howl
-import getmetatable, setfenv, pairs, callable, print, tostring, pcall from _G
+import getmetatable, setfenv, pairs, callable, print, tostring, pcall, error from _G
```
Comment (shalabhc) [inline bundles/vi/state.moon]: 'error' unused?

**Comment by shalabhc** [comment]
Yup - looks great! :+1:

File: bundles/vi/state.moon (hunk 1 above)
Comment (nilnor) [inline bundles/vi/state.moon]: The opposite actually, a previously missing import.

**Pull request merged**